### ISSUE TITLE
colors only exist for default config

### ISSUE THREAD

**Issue by ORESoftware** [expressjs/morgan#161]
If I use:

```js
app.use(morgan('dev');
```

I see:

![screenshot 2018-02-27 16 33 39](https://user-images.githubusercontent.com/11139560/36763251-0374c964-1bdc-11e8-939c-393d59ba713a.png)

but when I use this:

```js
app.use(morgan(function (tokens, req, res) {
    return [
      tokens.method(req, res),
      decodeURI(tokens.url(req, res)),
      tokens.status(req, res),
      tokens.res(req, res, 'content-length'), '-',
      tokens['response-time'](req, res), 'ms'
    ]
    .join(' ')
  }));
```

I see this:

![screenshot 2018-02-27 16 37 38](https://user-images.githubusercontent.com/11139560/36763376-8bc42788-1bdc-11e8-83c3-d2631d652cea.png)

**Comment by dougwilson** [comment]
All the tokens are simply text, so when they are incorporated as part of a string and streamed into a database or file, there are none a bunch of escape characters in the text that are needed to colorize terminal text. The exception to this is the single `dev` format which will output color (and thus nor suitable to output into a file due to the escape sequences). This is done directly in the format function (https://github.com/expressjs/morgan/blob/master/index.js#L183-L206) by adding the escape sequences around the output of the standard `:status` token.

**Comment by ORESoftware** [comment]
thanks Doug

I frankly am not sure what to do with this:

```js
morgan.format('dev', function developmentFormatLine (tokens, req, res) {
  // get the status code if response written
  var status = headersSent(res)
    ? res.statusCode
    : undefined

  // get status color
  var color = status >= 500 ? 31 // red
    : status >= 400 ? 33 // yellow
    : status >= 300 ? 36 // cyan
    : status >= 200 ? 32 // green
    : 0 // no color

  // get colored function
  var fn = developmentFormatLine[color]

  if (!fn) {
    // compile
    fn = developmentFormatLine[color] = compile('\x1b[0m:method :url \x1b[' +
      color + 'm:status \x1b[0m:response-time ms - :res[content-length]\x1b[0m')
  }

  return fn(tokens, req, res)
})
```

**Comment by ORESoftware** [comment]
I guess you are using the cb function to cache the coloring functions, but I think it might be a little easier to figure out if you just used a local var?

**Comment by dougwilson** [comment]
I was just showing how the module is making the formatter and explain why your formatter is not colorized. In your function, you need to add the colors around the tokens you want to have colors, like what our dev format does.

> I guess you are using the cb function to cache the coloring functions, but I think it might be a little easier to figure out if you just used a local var?

I'm really sorry I'm not following on this. What is the cb function?

**Comment by ORESoftware** [comment]
`developmentFormatLine` is being used to cache the color fns, it's fine, just hard to follow what's going on immediately

**Comment by dougwilson** [comment]
The developmentFormatLine is the same as your anonymous function you have in your first post, just named.

**Comment by ORESoftware** [comment]
ok got it I can work with that thx

**Comment by dougwilson** [comment]
No problem 👍
